### PR TITLE
Move to Poetry and pyproject.toml

### PR DESCRIPTION
**Pull request by tomchop**
Poetry (https://python-poetry.org/) is yet another python dependency / virtualenv manager, but it provides significant advantages over pipenv:

* It can actually update single dependencies. This is something we do quite often with turbinia, grr, libcloudforensics, timesketch, e.g. https://github.com/log2timeline/dftimewolf/pull/635
  * pipenv can't, and probably won't be able to, do this https://github.com/pypa/pipenv/issues/2412, https://github.com/pypa/pipenv/pull/4988
* It can also take care of building and pushing to pypi (which we don't currently do, but that would be an option to explore). 

It can still generate requirements.txt files for people who want to fall back to just using pip.

Edit: I also upgraded `pylint`, which was quite outdated, and copied / pasted the log2timeline/plaso pylintrc file. We need to go over the error messages / codebase to see what warnings we want to keep and what new things to fix: https://github.com/log2timeline/dftimewolf/issues/637

### CODE REVIEW

**Review by sa3eed3ed — DISMISSED** [review]

**Comment by tomchop** [comment]
I'll hold off on @ramo-j's explicit LGTM on this before merging

**Comment by ramo-j** [comment]
I notice the update to pylint is causing quite a number of new linting errors. My preference would be to fix them in the same PR, but I notice this one is already `+6,521 −3,008` (Of course, mostly in lock+requirements files.) What do you reckon @tomchop?

**Comment by tomchop** [comment]
Sounds good, let's do that - I wanted to keep the number of changed files low for this one

File: .pylintrc
Comment (sa3eed3ed) [inline .pylintrc:5]: is this changed to master because you generated the file in a branch and that is the default? shouldn't this be the MAIN branch here?

Comment (tomchop) [reply .pylintrc:5]: This is from the .pylintrc file config, don't know why it was changed to MAIN before: https://www.codeac.io/documentation/pylint-configuration.html

**Pull request merged**
